File: DSA/Gautam_Shyamnani_590013078/Day43Question3.cpp
#include <iostream>
#include <queue>
#include <stack>
using namespace std;

bool isPalindrome(queue<char> q) {
    stack<char> s;
    int size = q.size();

    
    for (int i = 0; i < size; ++i) {
        char ch = q.front();
        s.push(ch);
        q.push(ch);     
        q.pop();
    }

    
    for (int i = 0; i < size; ++i) {
        char chQueue = q.front(); q.pop();
        char chStack = s.top(); s.pop();

        if (chQueue != chStack)
            return false;
    }

    return true;
}

int main() {
    queue<char> q1;
    string word = "racecar";  
    for (char ch : word) {
        q1.push(ch);
    }

    if (isPalindrome(q1))
        cout << "Yes, it's a palindrome!" << endl;
    else
        cout << "Nope, not a palindrome." << endl;

    return 0;
}

File: DSA/LakshaKundra_590013237/Day_43_Question_3.py
from collections import deque

def is_palindrome(q):
    stack = []
    for ch in q:
        stack.append(ch)
    for ch in q:
        if ch != stack.pop():
            return "No"
    return "Yes"

# Example 1
q1 = deque(['r', 'a', 'c', 'e', 'c', 'a', 'r'])
print(is_palindrome(q1))

# Example 2
q2 = deque(['a', 'b', 'c'])
print(is_palindrome(q2))
